### PR TITLE
Feat Wear OS Crashlytics

### PR DESCRIPTION
**Pull request by Andries-Smit**
Forward Wear OS crashes to phone, where they show in the log and register by Crashlytics

### CODE REVIEW

**Comment by MilosKozak** [comment]
I was trying to do the same but without success. This is working??

**Comment by Andries-Smit** [comment]
@MilosKozak sure! see crashlytics you will see a view "Testing Wear OS Crash ...." from various sources, the app, tile and activity.
image

There are two routes to log error, catch store in file system -> send on startup For example https://github.com/eBay/WearCrashReporter Or catch and send directly  https://gist.github.com/mauimauer/c6f40ec89863906e3b7a 
Any preference for the AAPS implementation?

File: app/src/main/java/info/nightscout/androidaps/plugins/general/wear/wearintegration/DataLayerListenerServiceMobile.kt [hunk 1]
```diff
@@ -115,12 +114,37 @@ class DataLayerListenerServiceMobile : WearableListenerService() {
...
                     val command = EventData.deserialize(String(messageEvent.data))
                     rxBus.send(command.also { it.sourceNodeId = messageEvent.sourceNodeId })
                 }
+
+                exceptionPath -> logWearException(messageEvent)
+            }
+        }
+    }
+
+    private fun logWearException(messageEvent: MessageEvent) {
```
Comment (MilosKozak) [inline app/src/main/java/info/nightscout/androidaps/plugins/general/wear/wearintegration/DataLayerListenerServiceMobile.kt:128]: I'd move this to FabricPrivacy and do not expose `instance()`

File: wear/src/main/java/info/nightscout/androidaps/comm/ExceptionService.kt [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        transcriptionNodeId?.also { nodeId ->
+            messageClient
+                .sendMessage(nodeId, path, data).apply {
+                    addOnSuccessListener { }
+                    addOnFailureListener {
+                        Log.d("WEAR", "sendMessage:  $path failure")
+                    }
+                }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (Andries-Smit) [reply wear/src/main/java/info/nightscout/androidaps/comm/ExceptionService.kt:58]: Sure it is duplication, I tried to re-use it, but I got trouble starting the service in the background.

The benefit of the code being independent allows us to log issues when there are issues in DataLayerListener too.

File: app/src/main/java/info/nightscout/androidaps/plugins/general/wear/wearintegration/DataLayerListenerServiceMobile.kt (hunk 1 above)
Comment (Andries-Smit) [reply app/src/main/java/info/nightscout/androidaps/plugins/general/wear/wearintegration/DataLayerListenerServiceMobile.kt:128]: When moving the whole function, it feels a bit dirty to pass MessageEvent to FabricPrivacy... not separation of concern?

Comment (MilosKozak) [reply app/src/main/java/info/nightscout/androidaps/plugins/general/wear/wearintegration/DataLayerListenerServiceMobile.kt:128]: you can decompile it here and then pass to FabricPrivacy as Exception

File: wear/src/main/java/info/nightscout/androidaps/comm/ExceptionService.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Andries-Smit) [reply wear/src/main/java/info/nightscout/androidaps/comm/ExceptionService.kt]: Update with re-use DataLayer and rx

**Comment by MilosKozak** [comment]
and still working?

**Comment by Andries-Smit** [comment]
> I was trying to do the same but without success. 
What failed?

**Pull request merged**